cmd/genesis/vmtracer.go

```diff
@@ -0,0 +1,38 @@
+package main
+
+import (
+	"github.com/kowala-tech/kcoin/common"
+	"github.com/kowala-tech/kcoin/core/vm"
+	"time"
+)
+
+//TODO:This vmTracer is an exact copy of the one in pupphet, can be unified?
+type vmTracer struct {
+	data map[common.Address]map[common.Hash]common.Hash
+}
+
+func newVmTracer() *vmTracer {
+	return &vmTracer{
+		data: make(map[common.Address]map[common.Hash]common.Hash, 1024),
+	}
+}
+
+func (vmt *vmTracer) CaptureState(env *vm.EVM, pc uint64, op vm.OpCode, gas, cost uint64, memory *vm.Memory, stack *vm.Stack, contract *vm.Contract, depth int, err error) error {
```

Comment:
I know this is copied from puppeth, but without tests this function is very confusing. What does it do?

Reply:
I think this is a tracer to capture some kind of information from the real blockchain, is there anybody that can assert this info?